src/cpp/session/modules/SessionConsole.R

```diff
@@ -14,3 +14,58 @@
 #
 #
 
+.rs.addJsonRpcHandler("console_follow_hyperlink", function(url, text, params)
+{
+
+  pattern <- "^([a-zA-Z]+)::(.*$)"
+  if (grepl(pattern, url)) {
+    target <- sub(pattern, "\\1", url)
+    url <- sub(pattern, "\\2", url)
+  } else {
+    target <- "browser"
+  }
+
+  handlers <- list(
+    # http://example.com
+    # browser::http://example.com
+    browser = function(url) utils::browseURL(url), 
+
+    # viewer::http://example.com
+    viewer = function(url) .rs.api.viewer(url),
+
+    # help::stats/rnorm
+    # help::stats::rnorm
+    help = function(url) {
+      parts <- strsplit(url, "[/:]+")[[1L]]
+
+      if (length(parts) == 2L) {
+        .rs.showHelpTopic(topic = parts[[2L]], package = parts[[1L]])
+      } else {
+        # help::rnorm
+        .rs.showHelpTopic(topic = parts[[1L]], package = NULL)
+      }
+    }, 
+
+    # file::some/file/path
+    # file::some/file/path#32
+    # file::some/file/path#32,15
+    file = function(url) {
+      parts <- strsplit(url, "#")[[1L]]
+      file <- parts[[1L]]
+
+      if (length(parts) == 2L) {
+        location <- strsplit(parts[[2L]], ",")[[1L]]
+        line <- as.numeric(location[[1L]])
+        if (length(location) == 1L) {
+          col <- -1L
+        } else {
+          col <- as.numeric(location[[2L]])
+        }
+      }
+      
+      .rs.api.navigateToFile(file, line = line, col = col, moveCursor = TRUE)
```

Comment:
This opens the file at the right location, but this does not seem to always move the focus. Or more accurately, it does but then the focus goes back to the console.